doc/generating-code.mld

```diff
@@ -0,0 +1,357 @@
+{%html: <div style="display: flex; justify-content:space-between"><div>%}{{!"writing-ppxs"}< Writing PPXs}{%html: </div><div>%}{{!"matching-code"}Destructing AST nodes >}{%html: </div></div>%}
+
+{0 Generating AST Nodes}
+
+The rewriter's core is a function that outputs code in the form of an AST.
```

Comment:
Unsure where, but I would add a link to astexplorer. If you are a first-timer here you would have a easier understanding once you see the parsetree as a tree and all the info there.

Reply:
Do you think that's necessary? (it is already mentioned in https://ocaml.org/docs/metaprogramming#ocamls-parsetree-the-ocaml-ast which is supposed to be a "prerequisite". The manual is targeted to authors of PPXs, who should have the idea of the AST as a tree.)

I could be added, but I feel it fits better in general explanations about AST/tutorial/good practices/recipes than in the manual. Maybe we can leave this improvement/new section for later?

Reply:
Yes, totally both options are great here. Another section to get deep into Parsetree and assuming readers to be aware of it works great